cmd/lotus-shed/export-car.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ipfs/go-blockservice"
+	"github.com/ipfs/go-cid"
+	offline "github.com/ipfs/go-ipfs-exchange-offline"
+	format "github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+	"github.com/ipld/go-car"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/filecoin-project/lotus/node/repo"
+)
+
+func carWalkFunc(nd format.Node) (out []*format.Link, err error) {
+	for _, link := range nd.Links() {
+		if link.Cid.Prefix().Codec == cid.FilCommitmentSealed || link.Cid.Prefix().Codec == cid.FilCommitmentUnsealed {
+			continue
+		}
+		out = append(out, link)
+	}
+	return out, nil
+}
+
+var exportCarCmd = &cli.Command{
+	Name:        "export-car",
+	Description: "Export a car from repo (requires node to be offline)",
```

Comment:
We could support online-mode with the API blockstore (`blockstore.NewAPIBlockstore`)

Reply:
Is there a good way to detect this?

Reply:
I guess do something like what we do in https://github.com/filecoin-project/lotus/blob/master/cli/wait.go

Could also just be an explicit --offline flag

Reply:
Eh, I'd rather not spend time on this for now.